src/components/Admin/AdminSearchForm.jsx

```diff
@@ -0,0 +1,163 @@
+/* eslint-disable camelcase */
+import qs from 'query-string';
+import React from 'react';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import { Form } from '@edx/paragon';
+
+import SearchBar from '../SearchBar';
+import { updateUrl } from '../../utils';
+
+class AdminSearchForm extends React.Component {
+  constructor(props) {
+    super(props);
+    const { location } = props;
+    const queryParams = qs.parse(location.search);
+    this.state = {
+      searchQuery: queryParams.search,
+      searchCourseQuery: queryParams.search_course,
+      searchDateQuery: queryParams.search_start_date,
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    const { location } = this.props;
+    if (location.search !== prevProps.location.search) {
+      // eslint-disable-next-line camelcase
+      const { search, search_course, search_start_date } = qs.parse(location.search);
+      const {
+        search: prevSearch,
+        search_course: prevSearchCourse,
+        search_start_date: prevSearchDate,
+      } = qs.parse(prevProps.location.search);
+      if (search !== prevSearch || search_course !== prevSearchCourse ||
+        search_start_date !== prevSearchDate) {
+        this.handleSearch(search, search_course, search_start_date);
+      }
+    }
+  }
+
+  onCourseSelect(event) {
+    const updateParams = {
+      search_course: event.target.value,
+      page: 1,
+    };
+    if (event.target.value === '') {
+      updateParams.search_start_date = '';
+    }
+    updateUrl(updateParams);
+  }
+
+  handleSearch(emailSearch, courseSearch, dateSearch) {
+    this.setState({
+      searchQuery: emailSearch,
+      searchCourseQuery: courseSearch,
+      searchDateQuery: dateSearch,
+    });
+    this.props.searchEnrollmentsList();
+  }
+
+  render() {
+    const { searchCourseQuery, searchDateQuery, searchQuery } = this.state;
+    const { tableData } = this.props;
+    const courseTitles = new Set(tableData.map(en => en.course_title));
+    const courseDates = new Set(tableData.map(en => en.course_start).sort().reverse());
+
+    return (
+      <div className="row">
+        <div className="col-12 pr-md-0 mb-0">
```

Comment:
wow the styling here seems to have gotten interesting. I am still trying to digest the various col > row gridness here ... maybe better once I have this UI locally

Reply:
It makes it look nice on mobile. 😉